linux/net/rina/dtcp.c

```diff
@@ -747,9 +1125,32 @@ static int rcv_flow_ctl(struct dtcp * dtcp,
         pci = pdu_pci_get_rw(pdu);
         ASSERT(pci);
 
-        snd_rt_wind_edge_set(dtcp, pci_control_new_rt_wind_edge(pci));
+        rt = pci_control_sndr_rate(pci);
+        tf = pci_control_time_frame(pci);
+
+        // Is this a window based control flow?
+        if(dtcp_window_based_fctrl(dtcp_config_get(dtcp))) {
+        	snd_rt_wind_edge_set(dtcp, pci_control_new_rt_wind_edge(pci));
+        	dt_sv_window_closed_set(dtcp->parent, false);
```

Comment:
That code was inherited from previous versions... and removed from you and Eduard the 25th August. Probably left behind due the long pause during end of August, first of September, when your commits were pushed.

I guess the correct part there should be:
```
snd_rt_wind_edge_set(dtcp, pci_control_new_rt_wind_edge(pci));
push_pdus_rmt(dtcp);
```

right?